extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -430,19 +437,76 @@ OHIFCornerstoneSRViewport.defaultProps = {
   customProps: {},
 };
 
+const findInstance = (measurement, displaySetService: DisplaySetService) => {
+  const {
+    displaySetInstanceUID,
+    ReferencedSOPInstanceUID: sopUid,
+  } = measurement;
+  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
+    displaySetInstanceUID
+  );
+  if (!referencedDisplaySet.images) return;
+  return referencedDisplaySet.images.find(it => it.SOPInstanceUID === sopUid);
+};
+
 async function _getViewportReferencedDisplaySetData(
   displaySet,
   measurementSelected,
   displaySetService
 ) {
   const { measurements } = displaySet;
-  const measurement = measurements[measurementSelected];
 
-  const { displaySetInstanceUID } = measurement;
+  if (!displaySet.keyImageDisplaySet) {
+    const findReferences = ds => {
+      const instances = [];
+      const instanceByUrl = {};
+      for (const measurement of ds.measurements) {
+        const instance = findInstance(measurement, displaySetService);
+        if (!instance) {
+          console.log('Measurement', measurement, 'had no instances found');
+          continue;
+        }
 
-  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
-    displaySetInstanceUID
-  );
+        // TODO - find actual URL
+        const { imageId } = measurement;
+        if (!imageId) continue;
+        if (instanceByUrl[imageId]) continue;
+        instanceByUrl[imageId] = instance;
+        instances.push(instance);
+      }
+      return instances;
+    };
+    const instances = findReferences(displaySet);
+    const updateInstances = function () {
+      this.images.splice(0, this.images.length, ...findReferences(displaySet));
```

Comment:
I don't like the new trend of having `this` all over the place, can we avoid it?

Reply:
When something is a member function, it should use this - I think this one is appropriate here.